presto-main/src/main/java/com/facebook/presto/operator/PartitionedOutputOperator.java

```diff
@@ -355,6 +371,7 @@ public PartitionedOutputInfo getInfo()
                     for (PageBuilder pageBuilder : pageBuilders) {
                         appendRow(pageBuilder, page, position);
                     }
+                    hasAnyRowBeenReplicated = true;
```

Comment:
This doesn't look correct. If `shouldReplicate` returned true due to a null, this will cause none of the non-null rows to be replicated.

Also, I don't like asymmetry and apparent side effect setting this has on `shouldReplicate`. I think it'd be easier to reason about if the code were inlined.